src/io.c

```diff
@@ -156,9 +157,22 @@ XcpError xcp_fd_pread (int fd, void *buf, size_t count, off_t offset) {
   return XCP_ERR_ERRNO;
 }
 
+XcpError xcp_fd_preadv (int fd, const struct iovec *iovs, size_t iovCount, off_t offset) {
+  do {
+    const ssize_t ret = preadv(fd, iovs, (int)iovCount, offset);
+    if (ret >= 0) return ret;
+
+  XCP_C_WARN_PUSH
+  XCP_C_WARN_DISABLE_LOGICAL_OP
+  } while (errno == EAGAIN || errno == EWOULDBLOCK || errno == EINTR);
+  XCP_C_WARN_POP
+
+  return XCP_ERR_ERRNO;
+}
+
 // -----------------------------------------------------------------------------
 
-XcpError xcp_poll (struct pollfd *fds, nfds_t nfds, int timeout) {
+XcpError xcp_poll (struct pollfd *fds, uint nfds, int timeout) {
   do {
     const int ret = poll(fds, nfds, timeout);
```

Comment:
Does the call build without casting nfds to `nfds_t` ?
IF it does we need to add an implicit cast warning i think.

Reply:
No compilation problem. I think it's useless to add a cast if there is no warning about this.